client/js/widgets/imagewidget.js

```diff
@@ -90,9 +93,23 @@ class ImageWidget extends Widget {
   }
 
   updateIcon() {
```

Comment:
Try to move the holder code into the Holder class by defining an `updateIcon` there that calls `super.updateIcon()`.